torch/jit/__init__.py

```diff
@@ -235,7 +262,10 @@ def forward(self, *args):
 # specified
 def _get_trace(f, args, in_vars, nderivs=0):
     trace = torch._C._tracer_enter(in_vars, nderivs)
+    global _tracing
+    _tracing = True
     out = f(*args)
+    _tracing = False
```

Comment:
Note: this is only called from `torch.jit.trace`. `torch.jit.compile` won't use scopes at all in this patch (which might not be such a bad thing - see the comment on optimizations)